source/common/http/filter/fault_filter.cc

```diff
@@ -69,7 +80,11 @@ FaultFilter::~FaultFilter() { ASSERT(!delay_timer_); }
 // if we inject a delay, then we will inject the abort in the delay timer
 // callback.
 FilterHeadersStatus FaultFilter::decodeHeaders(HeaderMap& headers, bool) {
-  if (!matchesTargetCluster()) {
+  if (!matchesTargetUpstreamCluster()) {
+    return FilterHeadersStatus::Continue;
+  }
+
+  if (!matchesDownstreamNodes(headers)) {
     return FilterHeadersStatus::Continue;
```

Comment:
Nevermind, talked again. This is kind of a far reaching change. To unblock let's go with this and we will add TODO to generalize and clean this up.